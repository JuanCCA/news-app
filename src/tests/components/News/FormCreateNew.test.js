import { shallow } from 'enzyme';
import FromCreateNew from '../../../components/News/FormCreateNew'
import React from 'react'
import '@testing-library/jest-dom';


describe('Testing <FromCreateNew />', () => {

    let wrapper = shallow(<FromCreateNew />)
    const handleSubmit = jest.fn()

    beforeEach( () => {
        jest.clearAllMocks()
        wrapper = shallow(<FromCreateNew />)
    })
    
    test('should create correctly the component', () => {

        expect(wrapper).toMatchSnapshot()

    })

    test('should show modal when button pressed', () => {

        wrapper.find('#form').simulate('submit')

        expect(handleSubmit).toHaveBeenCalled(); //TODO


    })


    

})
