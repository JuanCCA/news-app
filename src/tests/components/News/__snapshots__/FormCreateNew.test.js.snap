// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <FromCreateNew /> should create correctly the component 1`] = `
<Form
  id="form"
  onSubmit={[Function]}
>
  <FormGroup
    className="mb-3"
    controlId="formBasicEmail"
  >
    <FormLabel
      column={false}
      visuallyHidden={false}
    >
      Title
    </FormLabel>
    <FormControl
      onChange={[Function]}
      placeholder="Title"
      required={true}
      type="text"
      value=""
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
    controlId="description"
  >
    <FormLabel
      column={false}
      visuallyHidden={false}
    >
      Description
    </FormLabel>
    <FormControl
      onChange={[Function]}
      placeholder="Description"
      required={true}
      type="text"
      value=""
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
    controlId="content"
  >
    <FormLabel
      column={false}
      visuallyHidden={false}
    >
      Content
    </FormLabel>
    <FormControl
      as="textarea"
      onChange={[Function]}
      placeholder="Content"
      required={true}
      value=""
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
    controlId="author"
  >
    <FormLabel
      column={false}
      visuallyHidden={false}
    >
      Author
    </FormLabel>
    <FormControl
      onChange={[Function]}
      placeholder="Author"
      required={true}
      type="text"
      value=""
    />
  </FormGroup>
  <Button
    active={false}
    className="px-5"
    disabled={false}
    type="submit"
    variant="dark"
  >
    Save
  </Button>
</Form>
`;
